state-chain/pallets/cf-environment/src/tests.rs

```diff
@@ -92,6 +92,86 @@ fn test_btc_utxo_selection() {
 	});
 }
 
+#[test]
+fn test_btc_utxo_consolidation() {
+	new_test_ext().execute_with(|| {
+		let utxo = |amount| Utxo {
+			amount,
+			id: Default::default(),
+			deposit_address: DepositAddress::new(Default::default(), Default::default()),
+		};
+
+		// Reduce consolidation parameters to make testing easier
+		assert_ok!(Environment::update_consolidation_parameters(
+			OriginTrait::root(),
+			cf_chains::btc::ConsolidationParameters {
+				consolidation_threshold: 2,
+				consolidation_size: 2,
+			}
+		));
+
+		assert_eq!(
+			Environment::select_and_take_bitcoin_utxos(UtxoSelectionType::SelectForConsolidation),
+			None
+		);
+
+		let dust_amount = {
+			use cf_traits::GetBitcoinFeeInfo;
+			<Test as crate::Config>::BitcoinFeeInfo::bitcoin_fee_info().fee_per_input_utxo
+		};
+
+		add_utxo_amount(10000);
+		// Some utxos exist, but it won't be enough for consolidation:
+		assert_eq!(
+			Environment::select_and_take_bitcoin_utxos(UtxoSelectionType::SelectForConsolidation),
+			None
+		);
+		assert_eq!(crate::BitcoinAvailableUtxos::<Test>::decode_len(), Some(1));
+
+		// Dust utxo does not count:
+		add_utxo_amount(dust_amount);
+		assert_eq!(
+			Environment::select_and_take_bitcoin_utxos(UtxoSelectionType::SelectForConsolidation),
+			None
+		);
+		assert_eq!(crate::BitcoinAvailableUtxos::<Test>::decode_len(), Some(2));
+
+		add_utxo_amount(10000);
+		add_utxo_amount(10000);
+
+		// Any utxo that didn't get consolidated should still be available:
+		assert_eq!(
+			Environment::select_and_take_bitcoin_utxos(UtxoSelectionType::SelectForConsolidation),
+			Some((vec![utxo(10000), utxo(10000)], 15980))
+		);
+		assert_eq!(crate::BitcoinAvailableUtxos::<Test>::decode_len(), Some(2));
```

Comment:
It would be good to ensure that the remaining utxos are the right ones... Maybe use different utxos, UTXO1, UTXO2, UTXO3, UTXO4, each with different amounts, and assert that the remaining and extracted together form the same set, ie. we didn't accidentally duplicate or delete any utxos.

Reply:
Sure. I will compare directly as `Vec`s since the logic is deterministic and to keep things simple.